lib/components/ClinicUserEdit.js

```diff
@@ -0,0 +1,234 @@
+  {value: '03', label: 'March'},
+  {value: '04', label: 'April'},
+  {value: '05', label: 'May'},
+  {value: '06', label: 'June'},
+  {value: '07', label: 'July'},
+  {value: '08', label: 'August'},
+  {value: '09', label: 'September'},
+  {value: '10', label: 'October'},
+  {value: '11', label: 'November'},
+  {value: '12', label: 'December'}
+];
+
+var ClinicUserEdit = React.createClass({
+  propTypes: {
+    errorMessage: React.PropTypes.string,
+    allUsers: React.PropTypes.object.isRequired,
+    loggedInUser: React.PropTypes.string.isRequired,
+    targetId: React.PropTypes.string,
+    updateUser: React.PropTypes.func.isRequired,
+    createUser: React.PropTypes.func.isRequired,
+    cancelEdit: React.PropTypes.func.isRequired
+  },
+
+  getInitialState: function(){
+    if(this.props.targetId){
+      var user = _.get(this.props.allUsers, this.props.targetId);
+      return {
+        name: _.get(user, 'fullName'),
+        year: _.get(user, ['patient', 'birthday'], '').substr(0,4),
+        month: _.get(user, ['patient', 'birthday'], '').substr(5,2),
+        day: _.get(user, ['patient', 'birthday'], '').substr(8,2),
+        email: _.get(user, ['patient', 'email'], ''),
+        mrn: _.get(user, ['patient', 'mrn'], '')
+      };
+    }
+    return {
+      name: '',
+      year: '',
+      month: '',
+      day: ''
+    };
+  },
+
+  zeroPad: function(value){
+    return (value.length === 1 ? '0': '') + value;
+  },
+
+  isDisabled: function() {
+    var dateString = _.get(this,'refs.year.value')+'-'+_.get(this,'refs.month.value')+'-'+this.zeroPad(_.get(this,'refs.day.value', ''));
+    return (!_.get(this, 'refs.name.value')) || (!_.get(this,'refs.month.value')) ||
+      (!_.get(this,'refs.day.value')) || (!_.get(this,'refs.year.value')) ||
+      (!sundial.isValidDateForMask(dateString, 'YYYY-MM-DD')) ||
+      (sundial.dateDifference(new Date(), dateString, 'd') < 0);
+  },
+
+  handleChange: function(e){
+    this.setState({[e.target.name]: e.target.value});
+  },
+
+  handleCancel: function(e){
+    this.props.cancelEdit();
+  },
+
+  handleNext: function(e) {
+    e.preventDefault();
+    this.setState({errorMessage:''});
+    var name = this.refs.name.value;
+    var dateString = this.refs.year.value+'-'+this.refs.month.value+'-'+this.zeroPad(this.refs.day.value);
+    var email = this.refs.email.value;
+    var mrn = this.refs.mrn.value;
+    if(sundial.isValidDateForMask(dateString, 'YYYY-MM-DD')){
+      var profile = {
+        fullName: name,
+        patient: {
+          birthday: dateString
+        }
+      };
+
+      if(email){
+        profile.patient.email = email;
+        profile.emails = [email];
+      }
+
+      if(mrn){
+        profile.patient.mrn = mrn;
+      }
+
+      if(this.props.targetId){
+        this.props.updateUser(profile);
+      } else {
+        this.props.createUser(profile);
+      }
+    } else {
+      this.setState({errorMessage: 'Invalid birthday.'});
```

Comment:
This won't ever surface because you're also checking `sundial.isValidDateForMask` in the `isDisabled` for the submit button...